tests/unit/api/c/dmx_merger/test_dmx_merger.cpp

```diff
@@ -75,15 +76,15 @@ class TestDmxMerger : public ::testing::Test
   // This determines what kind of merge test TestMerge does.
   enum class MergeTestType
   {
-    kUpdateSourceData,  // Merge using sacn_dmx_merger_update_source_data.
-    kStopSourcePap      // Merge using sacn_dmx_merger_update_source_data, then call
-                        // sacn_dmx_merger_stop_source_per_address_priority on the second source.
+    kUpdate,        // Merge using the sacn_dmx_merger_update functions.
+    kStopSourcePap  // Merge using the sacn_dmx_merger_update functions, then call
+                    // sacn_dmx_merger_remove_paps on the second source.
   };
 
-  void TestMerge(uint8_t priority_1, const uint8_t* values_1, uint16_t values_1_count,
-                 const uint8_t* address_priorities_1, uint16_t address_priorities_1_count, uint8_t priority_2,
-                 const uint8_t* values_2, uint16_t values_2_count, const uint8_t* address_priorities_2,
-                 uint16_t address_priorities_2_count, MergeTestType merge_type)
+  void TestMerge(const std::optional<int>& priority_1, const uint8_t* levels_1, uint16_t levels_1_count,
```

Comment:
This function has a ton of parameters, and feels like it's doing too many things. Does this really need to be a generic function / is the MergeTestType enum necessary? I've found that unit tests are one place where some repetition can be OK and can make the test code easier to understand. In a previous review we replaced this pattern with refactoring common setup code into its own setup function, then having the unique test code in each individual TEST_F() function.

Reply:
Let me know what you guys think about the direction I went with this: https://github.com/ETCLabs/sACN/pull/29/commits/b5ab10a11905fff9052a4a6bfe34fa44d284b34b